pcdsdevices/epics/areadetector/base.py

```diff
@@ -27,4 +27,12 @@ class ADComponent(base.ADComponent, Component):
 
 
 class ADBase(base.ADBase, Device):
-    pass
+    def stage(self, *args, **kwargs):
+        ret = Device.stage(self, *args, **kwargs)
+        try:
+            self.validate_asyn_ports()
+        except RuntimeError as err:
+            self.unstage(*args, **kwargs)
+            raise err
+        return ret 
```

Comment:
It does give you the stage method but the problem is that by default the stage method at ADBase does the following:

1. validate_asyn_ports
2. super().stage(...)

This patch allows us to change the nd_array_port before checking to see if the chain of plugins is valid. Before that it was not possible.